src/benchmarks/micro/libraries/System.Diagnostics/Perf_Activity.cs

```diff
@@ -0,0 +1,373 @@
+        private static readonly ActivitySource s_ActivitySource;
+        private static readonly ActivityListener s_ActivityListener;
+        private static readonly Activity s_Activity;
+        private static readonly ActivityLink s_ActivityLink;
+
+        static Perf_Activity()
+        {
+            s_ActivitySource = new ActivitySource("TestActivitySource");
+
+            s_ActivityListener = new ActivityListener
+            {
+                ShouldListenTo = s => s.Name == "TestActivitySource",
+
+                GetRequestedDataUsingContext = (ref ActivityCreationOptions<ActivityContext> o) => ActivityDataRequest.AllDataAndRecorded
+            };
+
+            ActivitySource.AddActivityListener(s_ActivityListener);
+
+            s_Activity = s_ActivitySource.StartActivity(
+                "TestActivity",
+                ActivityKind.Internal,
+                parentContext: default,
+                tags: new Dictionary<string, object>
+                {
+                    ["tag1"] = "string1",
+                    ["tag2"] = 1,
+                    ["tag3"] = "string2",
+                    ["tag4"] = false,
+                },
+                links: new ActivityLink[]
+                {
+                    new ActivityLink(default),
+                    new ActivityLink(default),
+                    new ActivityLink(default),
+                    new ActivityLink(default),
+                });
+
+            s_Activity.AddEvent(new ActivityEvent("TestEvent1"));
+            s_Activity.AddEvent(new ActivityEvent("TestEvent2"));
+            s_Activity.AddEvent(new ActivityEvent("TestEvent3"));
+            s_Activity.AddEvent(new ActivityEvent("TestEvent4"));
+
+            s_Activity.Stop();
+
+            s_ActivityLink = new ActivityLink(
+                default,
+                new ActivityTagsCollection(
+                    new Dictionary<string, object>
+                    {
+                        ["tag1"] = "string1",
+                        ["tag2"] = 1,
+                        ["tag3"] = "string2",
+                        ["tag4"] = false,
+                    }));
+        }
+
+        [Params(5000)]
+        public int NumberOfActivities { get; set; }
+
+        [Benchmark]
+        public void EnumerateActivityTags()
+        {
+            int total = 0;
+
+            for (int i = 0; i < NumberOfActivities; i++)
+            {
+                foreach (var tagString in s_Activity.Tags)
+                {
+                    total++;
+                }
+            }
+        }
+
+        [Benchmark]
+        public void EnumerateActivityTagObjects()
+        {
+            int total = 0;
+
+            for (int i = 0; i < NumberOfActivities; i++)
+            {
+                foreach (var tagObject in s_Activity.TagObjects)
+                {
+                    total++;
+                }
+            }
+        }
+
+        [Benchmark]
+        public void OTelHelperActivityTagObjects()
```

Comment:
>OTelHelperActivityTagObjects [](start = 20, length = 28)

could you please use the full word OpenTelemetry instead?

Reply:
Also, are we using OpenTelemetry  because this how it call the Activity for enumeration?

---
In reply to: [473440210](https://github.com/dotnet/performance/pull/1457#discussion_r473440210) [](ancestors = 473440210)

Reply:
Removed all the OpenTelemetry stuff.